portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -36,7 +36,7 @@
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Notice that since there's no user check here, this servlet allows anyone, including someone who is not logged in, to delete all comment data.

Reply:
I don't think any user could delete the comment data. Since this servlet only contains a doPost method, they can't access /delete-data (I checked and tried). The only way this servlet is called is with a button that only shows up if the user is logged in. I submitted changes without handling this case because I don't think it's possible, but I could be wrong - in which case, could you clarify/give examples of how they could delete? Thanks

Reply:
Note that there's nothing special about browsers or web forms. Anything the browser does when submitting a form, someone else could write a piece of software to do instead. Just because you don't show the form unless a user is logged in doesn't mean someone can't still post to it.

For example try this: with the server running in one terminal, open a new terminal and run:

`curl -X POST http://localhost:8000/delete-data`

(Note, check the port number, you might have to swap 8000 for 8080 or whatever the server is running as.)